### PR TITLE
Flesh out "SHOW COLUMNS" and "SHOW INDEX".

### PR DESCRIPTION
**Pull request by petermattis**
Added support for multi-column results.

### CODE REVIEW

File: sql/driver/conn.go [hunk 1]
```diff
@@ -213,6 +228,32 @@ func (c *conn) Select(p *parser.Select, args []driver.Value) (*rows, error) {
...
 
+func (c *conn) ShowColumns(p *parser.ShowColumns, args []driver.Value) (*rows, error) {
+	desc, err := c.getTableDesc(p.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	schema := structured.TableSchemaFromDesc(*desc)
+
+	// TODO(pmattis): This output doesn't match up with MySQL. Should it?
```
Comment (spencerkimball) [inline sql/driver/conn.go]: I'm sure someone somewhere depends on the format of this output, but unless all databases output the same thing, I wouldn't bother.

File: sql/driver/rows.go [hunk 2]
```diff
@@ -22,12 +22,24 @@ import (
...
 	columns []string
-	// TODO(pmattis): This is unfinished. Each row needs to have len(columns)
-	// entries.
-	rows []string
-	pos  int // Current iteration index into rows.
+	rows    []row
+	pos     int // Current iteration index into rows.
+}
+
+func newSingleColumnRows(column string, vals []string) *rows {
```
Comment (spencerkimball) [inline sql/driver/rows.go]: Honestly, I had to read the source to figure out what this is doing. The name is pretty inscrutable. Either come up with a better name or add a comment.

**Comment by spencerkimball** [comment]
LGTM

File: sql/driver/rows.go [hunk 3]
```diff
@@ -43,6 +58,8 @@ func (r *rows) Next(dest []driver.Value) error {
 	if r.pos >= len(r.rows) {
 		return io.EOF
 	}
-	dest[0] = r.rows[r.pos]
+	for i, v := range r.rows[r.pos] {
```
Comment (tschottdorf) [inline sql/driver/rows.go]: check for `len(dest) >= len(r.rows[r.pos])`?

**Comment by tschottdorf** [comment]
LGTM

File: sql/driver/rows.go (hunk 3 above)
Comment (petermattis) [inline sql/driver/rows.go]: `len(dest)` is guaranteed to be `len(r.Columns())`: https://golang.org/pkg/database/sql/driver/#Rows

        // Next is called to populate the next row of data into
        // the provided slice. The provided slice will be the same
        // size as the Columns() are wide.
        //
        // The dest slice may be populated only with
        // a driver Value type, but excluding string.
        // All string values must be converted to []byte.
        //
        // Next should return io.EOF when there are no more rows.

**Pull request merged**